packages/core/src/common/abstractPureComponent.ts

```diff
@@ -30,14 +30,14 @@ export abstract class AbstractPureComponent<P, S = {}> extends React.PureCompone
 
     constructor(props?: P, context?: any) {
         super(props, context);
-        if (!isNodeEnv("production")) {
+        if (!isNodeEnv('production')) {
             this.validateProps(this.props);
         }
     }
 
-    public componentWillReceiveProps(nextProps: P & { children?: React.ReactNode }) {
-        if (!isNodeEnv("production")) {
-            this.validateProps(nextProps);
+    public componentDidUpdate(_: P, __: S, ___: SS) {
+        if (!isNodeEnv('production')) {
+            this.validateProps(this.props);
```

Comment:
why is this not using `nextProps` anymore?

Reply:
`componentWillReceiveProps` provides the next set of props as argument, but `componentDidUpdate` provides the previous props. we want to validate the props that have just been applied, not the old ones